src/extensionConflicts.ts

```diff
@@ -0,0 +1,69 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved..
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { commands, Extension, extensions, window } from 'vscode';
+
+// A set of VSCode extension ID's that conflict with our extension
+const conflictingIDs = new Set([
+    'haaaad.ansible',
+    'sysninja.vscode-ansible-mod',
+    'tomaciazek.ansible',
+    'vscoss.vscode-ansible',
+    'zbr.vscode-ansible',
+]);
+
+// A set of VSCode extension ID's that are currently uninstalling
+const uninstallingIDs = new Set();
+
+/**
+ * Get all of the installed extensions that currently conflict with VSCode-YAML
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function getConflictingExtensions(): Extension<any>[] {
+  const conflictingExtensions: Extension<any>[] = [];
+  conflictingIDs.forEach((extension) => {
+    const ext = extensions.getExtension(extension);
+    if (ext && !uninstallingIDs.has(ext.id)) {
+      conflictingExtensions.push(ext);
+    }
+  });
+  return conflictingExtensions;
+}
+
+/**
+ * Display the uninstall conflicting extension notification if there are any conflicting extensions currently installed
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function showUninstallConflictsNotification(conflictingExts: Extension<any>[]): void {
+  // Add all available conflicting extensions to the uninstalling IDs map
+  for (const extIndex in conflictingExts) {
+    const ext = conflictingExts[extIndex];
+    uninstallingIDs.add(ext.id);
+  }
+
+  const uninstallMsg = 'Uninstall';
+
+  // Gather all the conflicting display names
+  let conflictMsg = '';
+  if (conflictingExts.length === 1) {
+    conflictMsg = `${conflictingExts[0].packageJSON.displayName} extension is incompatible with VSCode-YAML. Please uninstall it.`;
+  } else {
+    const extNames: string[] = [];
+    conflictingExts.forEach((ext) => {
```

Comment:
Plz use `map()` here.

Reply:
The entire code was copied from yaml extension and I only made the minimum number of changes in order to pass our linting. I do not want to diverge more as it would make harder to backport changes from there.

Reply:
Then it should go in a special `_vendor` location and never be linted.

Reply:
I implemented the changes you requested so vendoring is no longer needed.